knowledge_verificator/command_line.py

```diff
@@ -0,0 +1,135 @@
+        user_choice = choose_from_menu(
+            menu_elements=options, plural_name='options'
+        )
+
+        match user_choice:
+            case 'knowledge database':
+                try:
+                    material_db = MaterialDatabase(config.learning_materials)
+                except FileNotFoundError:
+                    console.print(
+                        f'In the `{config.learning_materials}` there is no database. '
+                        'Try using your own materials.'
+                    )
+                    continue
+
+                if not material_db.materials:
+                    console.print(
+                        'The knowledge database exists but is empty. '
+                        'Try using your own materials.'
+                    )
+                    continue
+
+                material = choose_from_menu(
+                    material_db.materials,
+                    plural_name='materials',
+                    attribute_to_show='title',
+                )
+
+                if material is None:
+                    continue
+
+                paragraph = str(
+                    choose_from_menu(material.paragraphs, 'paragraphs')
+                )
+
+                if paragraph is None:
+                    continue
+
+                console.print('Learn this paragraph: ')
+                console.print(paragraph)
+                console.print()
+                input('Press ENTER when ready.')
+
+            case 'my own paragraph':
+                console.print('Enter a paragraph you would like to learn: ')
+                paragraph = input().strip()
+
+            case _:
+                console.print('Unrecognised option, try again!')
+
+        logger.debug('Loaded the following paragraph:\n %s', paragraph)
+
+        chosen_answer = chooser.choose_answer(paragraph=paragraph)
+        if not chosen_answer:
+            logger.error(
+                'The supplied paragraph is either too short or too general. '
+                'Please, try providing a longer or more specific paragraph.'
+            )
+            continue
+
+        console.clear()
+
+        logger.debug(
+            'The `%s` has been chosen as the answer, based on which question '
+            'will be generated.',
+            chosen_answer,
+        )
+
+        question_with_context = qg_module.generate(
+            answer=chosen_answer, context=paragraph
+        )
+        question = question_with_context['question']
+        logger.debug(
+            'Question Generation module has supplied the question: %s', question
+        )
+
+        console.print(
+            f'\nAnswer the question with full sentence. {question} \nYour answer: '
+        )
+        user_answer = input().strip()
+        relation = nli_module.infer_relation(
+            premise=paragraph, hypothesis=user_answer
+        )
+
+        display_feedback(relation=relation, chosen_answer=chosen_answer)
```

Comment:
`relation=`, `chosen_anwser=` are bit redundant

Reply:
They are totally optional but my rationale is: if a arguments' order changes (this is early version so code changes are often), I do not have to modify this line. 

I understand it may look redundant and strange but I would prefer to preserve it as it is more coherent with the rest of the codebase.

Perhaps, is there another reason? I am open to discussion.